torch/csrc/autograd/functions/tensor.cpp

```diff
@@ -139,7 +140,6 @@ CopySlices::CopySlices(const Variable& base_var, TensorGeometry view, std::share
   // to base instead of the view.
   next_functions.resize(fn->next_functions.size());
   next_functions[0] = std::make_pair(base_var.grad_fn(), base_var.output_nr());
-  fn->next_functions[0] = next_functions[0];
```

Comment:
Won't this cause `fn->next_functions[0].first` to be a NULL pointer? This would cause `should_compute_output(0)` to return `false`, which might be incorrect

Reply:
No, the fn->next_functions are set correctly before it's passed to CopySlices. For example, `elu_` looks something like:

```c++
Tensor VariableType::elu_(Tensor & self, Scalar alpha) const {
    profiler::RecordFunction profiler("elu_");
    auto& self_ = unpack(self, "self", 0);
    check_inplace(self);
    std::shared_ptr<EluBackward> grad_fn;
    auto flags = compute_flags({ self });
    if (flags.requires_grad) {
      grad_fn = std::make_shared<EluBackward>();
      grad_fn->is_executable = true;
      grad_fn->next_functions = compute_next_functions({ self });
      grad_fn->alpha = alpha;
    }
   ...
}
```

The line `grad_fn->next_functions = ...` is the relevant one here.

If "self" is a view, then next_functions will point to something like AsStridedBackward or ViewBackward, which is what we want.

The test `test_elu_inplace_view` and `test_relu_inplace_view` tests this, since they save inputs or outputs. If the grad_fn's are incorrect, the double-backwards tests fail, which was previously the case.